### PR TITLE
Playerbot fury warrior improvements

### PR DESCRIPTION
**Pull request by mikehackworth**
## 🍰 Pullrequest
Improvements to Fury Warrior DPS when running a Fury spec.
- Enabling the use of Rampage, Sweeping Strikes, Improved Demoralizing Shout and Cleave 
- Reduce Heroic Strike spamming to save enough rage for primary rotation (BT, WW) and maintaining Rampage, only use it when dumping rage (rage > 60).

### Proof
- None

### Issues
- None

### How2Test
- using a build of mangosd based off master branch with playerbot enabled (BUILD_PLAYERBOT=1)
- Spec a warrior character into the Fury tree (as far as Rampage)
- .bot add <above character name>
- .bot orders combat assist
- engage in some combat and observe use of abilities in combat log and optionally measure Damage done using a Damage Meter Interface addon such as Recount
- repeat above steps using a build of mangosd based off this branch also with playerbot enabled and note the use of Rampage, Sweeping Strikes  and overall increase in damage done if using a Damage Meter.

### Todo / Checklist
- [X] None

### CODE REVIEW

File: src/game/PlayerBot/AI/PlayerbotWarriorAI.cpp [hunk 1]
```diff
@@ -337,10 +360,19 @@ CombatManeuverReturns PlayerbotWarriorAI::DoNextCombatManeuverPVE(Unit* pTarget)
...
             return RETURN_CONTINUE;
-        if (SLAM > 0 && m_ai.CastSpell(SLAM, *pTarget) == SPELL_CAST_OK)
+        if (SWEEPING_STRIKES > 0 && m_ai.GetAttackerCount() > 1 && !m_bot.HasAura(SWEEPING_STRIKES, EFFECT_INDEX_0) && m_bot.IsSpellReady(SWEEPING_STRIKES) && m_ai.CastSpell(SWEEPING_STRIKES) == SPELL_CAST_OK)
             return RETURN_CONTINUE;
-        if (HEROIC_STRIKE > 0 && m_ai.CastSpell(HEROIC_STRIKE, *pTarget) == SPELL_CAST_OK)
+        if (IMPROVED_DEMORALIZING_SHOUT > 0 && DEMORALIZING_SHOUT > 0 && !pTarget->HasAura(DEMORALIZING_SHOUT, EFFECT_INDEX_0) && m_ai.CastSpell(DEMORALIZING_SHOUT, *pTarget) == SPELL_CAST_OK)
             return RETURN_CONTINUE;
+        if (SLAM > 0 && IMPROVED_SLAM > 0 && m_ai.CastSpell(SLAM, *pTarget) == SPELL_CAST_OK)
+            return RETURN_CONTINUE;
+        if (m_ai.GetRagePercent() > 60 )
```
Comment (mikehackworth) [inline src/game/PlayerBot/AI/PlayerbotWarriorAI.cpp:369]: Despite being used elsewhere already m_ai.GetRageAmount() seems to return values that do not match the actual rage of the bot as observed in the GUI and combat log and often far exceeds 100. So I added GetRagePercent() which works as expected here.

Comment (killerwife) [reply src/game/PlayerBot/AI/PlayerbotWarriorAI.cpp:369]: Rage amount is from 0 to 1000 afaik. Not 0 to 100. At least if i remember right.

**Comment by cala** [comment]
Thank you for this PR. I will review it.

**Comment by mikehackworth** [comment]
Hi @cala just checking in on this to see if you have any feedback?

**Comment by cala** [comment]
It just went under my radar, sorry. I had an awefull end of the year. I'll try to handle this shortly.

**Comment by cala** [comment]
Thank you very much for this PR. It is merged in https://github.com/cmangos/mangos-tbc/commit/168297352500275b4afe6254737020b539c7e9cf

I made two small changes:
- I dropped the `PlayerBotAI::GetRagePercent()` method that you've added in favor of `Unit::GetPowerPercent()`. The various `GetPercent*` methods we have in PlayerBot code were added there before similar methods were made available in `Unit`. I intend to remove them somewhere in the future and use the ones in `Unit` instead as I want PlayerBot code to have as little custom code as possible and to rely more on core official methods.
- I removed the target when casting Demoralizing Shout because this spell does not require a target and that was causing the cast to fail

Thanks again for your contribution. 👍🏻

**Pull request closed without merging**